### PR TITLE
[RFID-03] Feat rfid pincode

### PR DESCRIPTION
**Pull request by Valyssa**
Added reading and writing to cards, improved keypad input

### CODE REVIEW

File: modules/RFID/CMakeLists.txt [hunk 1]
```diff
@@ -9,21 +9,30 @@ enable_testing()
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 
+#set(ldflags
+#        "-lbcm2835")
+#
+#string(REGEX REPLACE ";" " " ldflags "${ldflags}")
+#set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${LDflags}")
```
Comment (philippezwietering) [inline modules/RFID/CMakeLists.txt]: Remove this commented out area

File: modules/RFID/src/config-file-parser.cc [hunk 2]
```diff
@@ -65,26 +64,26 @@ void ConfigFileParser::loadDatabaseSettings(std::string & ip, std::string & user
...
+//    incorrectLineCheck(input != "KEY:");
+//    file >> encryptionKey;
+//    incorrectLineCheck(file.eof());
+//
+//    // throw error if reading went wrong
+//    if (!encryptionKey) {
+//        throw std::runtime_error("Missing input after reading file: ");
+//    }
+//    file.close();
+//}
```
Comment (philippezwietering) [inline modules/RFID/src/config-file-parser.cc]: Remove this commented out function

File: modules/RFID/src/config-file-parser.hh [hunk 3]
```diff
@@ -77,19 +76,19 @@ public:
...
+//     *
+//     * Typical usage:
+//     * \code
+//     *     int encryptionKey;
+//     *     loadDatabaseSettings(encryptionKey);
+//     * \endcode
+//     *
+//     * \param [out] encryptionKey The encryption key to be read from the file
+//     */
+//    void loadEncryptionSettings(int & encryptionKey);
```
Comment (philippezwietering) [inline modules/RFID/src/config-file-parser.hh]: Remove this commented out section.

File: modules/RFID/src/main.cc [hunk 4]
```diff
@@ -1,96 +1,149 @@
...
 #include "databasemanager.hh"
 
 #include <wiringPi.h>
 #include <wiringPiSPI.h>
- 
+
+#include <iomanip>
 #include <iostream>
 
+struct MFAuthentData {
```
Comment (philippezwietering) [inline modules/RFID/src/main.cc]: This struct isn't used in the rest of the file, please remove it.

**Review by philippezwietering — CHANGES_REQUESTED** [review]
Please remove unnecessary comments, like deprecated functions and whatnot. 
Also don't forget to add yourself to the author field

File: modules/RFID/CMakeLists.txt (hunk 1 above)
Comment (JeremyRuizenaar) [reply modules/RFID/CMakeLists.txt]: ^^

File: modules/RFID/src/config-file-parser.cc (hunk 2 above)
Comment (JeremyRuizenaar) [reply modules/RFID/src/config-file-parser.cc]: ^^

File: modules/RFID/src/config-file-parser.hh (hunk 3 above)
Comment (JeremyRuizenaar) [reply modules/RFID/src/config-file-parser.hh]: ^^

File: modules/RFID/src/main.cc [hunk 5]
```diff
@@ -1,96 +1,149 @@
...
-        
-        ConfigFileParser factory("database-config.txt", "encryption-config.txt");
 
+        //int encryptionKey;
+
+        ConfigFileParser factory("database-config.txt");
         factory.loadDatabaseSettings(ip, username, password);
-        factory.loadEncryptionSettings(encryptionKey);
 
-        databasemanager database;
```
Comment (JeremyRuizenaar) [inline modules/RFID/src/main.cc:35]: the datamanager database can be removed

File: modules/RFID/src/main.cc [hunk 6]
```diff
@@ -1,96 +1,149 @@
...
+        MySql connection;
 
-        database.connectTo(ip, username, password);
```
Comment (JeremyRuizenaar) [inline modules/RFID/src/main.cc]: can be removed because it's unused

File: modules/RFID/src/main.cc [hunk 7]
```diff
@@ -1,96 +1,149 @@
...
-        database.selectDatabase("R2D2");
+        connection.connectTo(ip, username, password);
+        connection.selectDatabase("R2D2");
```
Comment (JeremyRuizenaar) [inline modules/RFID/src/main.cc]: can be removed

File: modules/RFID/src/main.cc [hunk 8]
```diff
@@ -1,96 +1,149 @@
...
 
-            std::cout << c << " key has been pressed\n";
+#endif
 
-            std::cout << "Enter a pincode, ending with '#'\n";
+            rfid.PCD_StopCrypto1();
 
-            std::string pin = keypad.getString();
-            std::cout << "The pin you entered was: " << pin << "\n";
+            //connection.executeQuery("SELECT * FROM RFID");
```
Comment (JeremyRuizenaar) [inline modules/RFID/src/main.cc]: this query is unused. can be removed

File: modules/RFID/src/main.cc [hunk 9]
```diff
@@ -1,96 +1,149 @@
...
-                      << database.getAllCardIdFromDatabase()
-                      << '\n';
-            
-            std::cout << "String before encryption: R2D2 project\n";
-            std::cout << "String after encryption: "
-                      << encryption.Encrypt("R2D2 project")
-                      << '\n';
+//            std::cout << "Database information: "
+//                      << database.getAllCardIdFromDatabase()
+//                      << '\n';
```
Comment (JeremyRuizenaar) [inline modules/RFID/src/main.cc]: comments can be removed

**Review by philippezwietering — APPROVED** [review]

**Pull request merged**